java/benchmarks/src/main/java/glide/benchmarks/utils/Benchmarking.java

```diff
@@ -107,6 +114,23 @@ public static void printResults(
     public static void testClientSetGet(
             Supplier<Client> clientCreator, BenchmarkingApp.RunConfiguration config, boolean async) {
         for (int concurrentNum : config.concurrentTasks) {
+            // same as Executors.newCachedThreadPool() with a RejectedExecutionHandler for robustness
+            ExecutorService executor =
+                    new ThreadPoolExecutor(
+                            0,
+                            Integer.MAX_VALUE,
```

Comment:
Shouldn't we keep some soft limits, like 4xN CPUS?

Reply:
This is what the defaults are using.  Should we limit this?